sycl/doc/extensions/LevelZeroBackend/LevelZeroBackend.md

```diff
@@ -0,0 +1,126 @@
+|``` make<queue>(const context &, ze_command_queue_handle_t);```| Constructs a SYCL queue instance from a Level-Zero ```ze_command_queue_handle_t```. The context argument must be a valid SYCL context encapsulating a Level-Zero context. The queue is attached to the first device in the passed SYCL context.|
+|``` make<program>(const context &, ze_module_handle_t);```| Constructs a SYCL program instance from a Level-Zero ```ze_module_handle_t```. The context argument must be a valid SYCL context encapsulating a Level-Zero context. The Level-Zero module must be fully linked (i.e. not require further linking through [```zeModuleDynamicLink```](https://spec.oneapi.com/level-zero/latest/core/api.html?highlight=zemoduledynamiclink#_CPPv419zeModuleDynamicLink8uint32_tP18ze_module_handle_tP28ze_module_build_log_handle_t)), and thus the SYCL program is created in the "linked" state.|
+
+NOTE: We shall consider adding other interoperability as needed, if possible.
+                
+### 4.4 Level-Zero handles' ownership and thread-safety
+        
+The Level-Zero runtime doesn't do reference-counting of its objects, so it is crucial to adhere to these
+practices of how Level-Zero handles are manged.
+                
+#### 4.4.1 SYCL runtime takes ownership
+                
+Whenever the application creates a SYCL object from the corresponding Level-Zero handle via one of the ```make<T>()``` functions,
+the SYCL runtime takes ownership of the Level-Zero handle. The application must not use the Level-Zero handle after
+the last host copy of the SYCL object is destroyed (as described in the core SYCL specification under
+"Common reference semantics"), and the application must not destroy the Level-Zero handle itself.
+                                                                
+#### 4.4.2 SYCL runtime assumes ownership
+
+The application may call the ```get_native<T>()``` member function of a SYCL object to retrieve the underlying Level-Zero handle,
+however, the SYCL runtime continues to retain ownership of this handle. The application must not use this handle after
+the last host copy of the SYCL object is destroyed (as described in the core SYCL specification under
+"Common reference semantics"), and the application must not destroy the Level-Zero handle.
+
+#### 4.4.3 Considerations for multi-threaded environment
+
+The Level-Zero API is not thread-safe, but it is free-threaded meaning that only the access to its global/shared state is guarded for thread-safety.
+Applications must make sure that the Level-Zero handles themselves aren't used simultaneously from different threads.
+Practically speaking, and taking into account that SYCL runtime takes ownership of the Level-Zero handles,
+the application should not attempt further direct use of those handles.

```

Comment:
This last statement, of course, calls into question the utility of the `get_native()` member functions.  We say above that the application cannot use the handle returned by `get_native()` after the SYCL object is destroyed.  Now, we are saying that the application can't use the handle before the object is destroyed either!

Reply:
The handle returned by the get_native() can be used, but it will compete with SYCL RT. There are certainly ways to synchronize between the two (e.g. using a shared lock)

Reply:
How can the application use a shared lock?  The SYCL RT may have an internal lock it uses to access the Level-Zero handles, but this lock isn't exposed to the application (is it?).  If it is not exposed to the application, it can't be a shared lock.

Reply:
The internal lock is not exposed, of course. But SYCL constructs can wait on an application lock.